cmd/snap/cmd_set_test.go

```diff
@@ -127,6 +127,97 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
+func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
```

Comment:
```suggestion
func (s *snapSetSuite) TestSnapSetJSONValidationUnhappy(c *check.C) {
```

Reply:
As mentioned, would be great to have a parallel `TestSnapSetJSONValidationHappy` which checks valid raw json (as opposed to parsed below).